src/main/java/com/store/book/security/BasicAuthenticationFilter.java

```diff
@@ -0,0 +1,63 @@
+package com.store.book.security;
+
+import com.store.book.exception.BadCredentialsException;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpFilter;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+@Component
+@RequiredArgsConstructor
+public class BasicAuthenticationFilter extends HttpFilter {
```

Comment:
Please don’t complicate your solution for now. You can see how to do it in the best way in the next lessons